### ISSUE TITLE
[macOS] The state should be changed to AFK when loginwindow is the current window

### ISSUE THREAD

**Issue by naturecodevoid** [ActivityWatch/aw-watcher-afk#66]
I suggest that whenever loginwindow is the current window on macOS, the state be changed to AFK. loginwindow is what is shown when the computer is locked, which means the user is away from their computer. This can be observed with aw-watcher-window.

**Comment by naturecodevoid** [comment]
A better way to do this would be to check the `CGSSessionScreenIsLocked` key of the session info dictionary. See this StackOverflow answer for more info: https://stackoverflow.com/a/11511419

There is also a pure bash answer, if the Quartz library is not available (it wasn't when I tried): https://stackoverflow.com/a/66723000

**Comment by 0xbrayo** [comment]
The entire period is tracked as active except for a few times the afk watcher emitted events. Inferring afk state from loginwindow or `CGSSessionScreenIsLocked` seems like the way to go <img width="1535" height="370" alt="Image" src="https://github.com/user-attachments/assets/25aed8b3-c45a-420d-8d2e-b1e19d3c740c" />

**Comment by amsam0** [comment]
> Can be accessed via the private SkyLight framework 

This is not true. All you need is CoreFoundation and CoreGraphics. Any human would be able to tell this due to the name of the function starting with CG.

**Comment by TimeToBuildBob** [comment]
Fair correction — you're right that you don't need to link `SkyLight.framework` directly. `CGSessionCopyCurrentDictionary()` (and the `CGSSessionScreenIsLocked` key it returns) is reachable with just an `extern` declaration linked against `CoreGraphics`/`ApplicationServices`; the symbol resolves at runtime since those umbrella frameworks depend on `SkyLight` privately. Doesn't change the recommendation — option 2 (distributed notifications for lock/unlock) is still the cleanest fit for an event-driven watcher, with `CGSessionCopyCurrentDictionary` as a fallback/poll check.
